### DISCUSSION TITLE
Layout with multiple outpus: table in margin next to a figure

### DISCUSSION THREAD

**Discussion by mvwestendorp** [quarto-dev/quarto-cli#1201] [Q&A]
Is it possible to switch the figure and table in this [example](https://quarto.org/docs/authoring/article-layout.html#multiple-outputs)?  That is, the table in the margin and figure in the body.

Using `#| tbl-column: margin` does not seem to work in this case.

**Comment by dragonstyle** [comment]
It should be just that easy (and it seemed like it worked for me using the following):

```
---
title: test
format: html
---

```{r}
#| tbl-column: margin
#| echo: false
#| warning: false

mtcars2 <- mtcars
mtcars2$am <- factor(
  mtcars$am, labels = c('automatic', 'manual')
)

knitr::kable(
  mtcars[1:6, 1:6]
)

library(ggplot2)
ggplot(mtcars2, aes(hp, mpg, color = am)) +
  geom_point() +  geom_smooth(formula = y ~ x, method = "loess") +
  theme(legend.position = 'bottom')
```

![Screen Shot 2022-06-22 at 12 24 37 PM](https://user-images.githubusercontent.com/261654/175085275-3d07d860-02b2-473f-a3d9-fe94c0b190b6.png)

The table doesn't look great in this case because of its width, but this should work just fine...

**Reply by mvwestendorp** [reply]
Thanks for your reply! That was what I was expecting. Your test document does indeed work for me.

However, I still have the issue in my Quarto book. 
I think it may have to do with the use of captions though since the following example reflects what I see happening.

```
---
title: test
format: html
---


@fig-test

```{r}
#| label: fig-test
#| fig-cap: "test figure"
#| tbl-cap: "test table"
#| tbl-column: margin
#| echo: false
#| warning: false

mtcars2 <- mtcars
mtcars2$am <- factor(
  mtcars$am, labels = c('automatic', 'manual')
)

knitr::kable(
  mtcars[1:6, 1:6]
)

library(ggplot2)
ggplot(mtcars2, aes(hp, mpg, color = am)) +
  geom_point() +  geom_smooth(formula = y ~ x, method = "loess") +
  theme(legend.position = 'bottom')
```
```

![image](https://user-images.githubusercontent.com/902633/175251057-34c9afec-7ba1-48e0-a7d1-4825c403a8b2.png)

**Reply by dragonstyle** [reply]
The issue is the fact that the chunk has a label (`fig-*`). This is causing the outputs of the chunk (including the table) to be treated as figures, meaning that the table is being labeled using the `fig-cap`, being called `Figure 1`, and getting the `fig-column` rather than the `tbl-column`. 

I wasn't really able to create a chunk that could do this while also being cross reference-able (since we basically don't support differing types of outputs from the same chunk in the cross reference context). Placing them in separate chunks will allow the options to work as expected, but probably isn't what you're going for since the table doesn't end up in the margin next to the plot, but instead in the margin below the plot.

**Reply by mvwestendorp** [reply]
I understand and your suggestion is indeed my current 'workaround'.

Is "differing types of outputs from the same chunk in the cross reference context" something that could be supported in quarto or would that be too complex for the relatively rare use-case?

**Reply by dragonstyle** [reply]
I think layouts like this make a case for trying to support this- we're in the process of stabilizing our 1.0 release so we wouldn't do this now, but it is something we could add in the future. I'll open an issue referencing this discussion to track that.

**Reply by dragonstyle** [reply]
One other suggestion that could provide a slight improvement- you could place the outputs of two code chunks in a layout side by side with appropriate column widths:

https://quarto.org/docs/authoring/figures.html#complex-layouts

and then give that layout an overflowing column using:

https://quarto.org/docs/authoring/article-layout.html#overflowing-content

I haven't tried this, but it might be an improvement...

**Reply by mvwestendorp** [reply]
> I think layouts like this make a case for trying to support this- we're in the process of stabilizing our 1.0 release so we wouldn't do this now, but it is something we could add in the future. I'll open an issue referencing this discussion to track that.

Great, I would be interested in that addition to Quarto!

> One other suggestion that could provide a slight improvement- you could place the outputs of two code chunks in a layout side by side with appropriate column widths:
> 
> https://quarto.org/docs/authoring/figures.html#complex-layouts
> 
> and then give that layout an overflowing column using:
> 
> https://quarto.org/docs/authoring/article-layout.html#overflowing-content
> 
> I haven't tried this, but it might be an improvement...

If there is a short text before the figure the result is "good enough" for my purposes since the table will be approximately next to the figure.

**Reply by jhchou** [reply]
> One other suggestion that could provide a slight improvement- you could place the outputs of two code chunks in a layout side by side with appropriate column widths:

I was searching for a solution of trying to ensure my `::: {.column-margin}` explanatory note was strictly associated with specific text in the body, because every once in a while, the margin note would get placed elsewhere, resulting in a confusing "explanation" to a random other chunk of text..

While it could be "forced" with this side-by-side layout, I suspect getting the correct column widths to match up would fail if, for example, the size of the HTML viewport changed.

Any possibility of building in a way to automate generating this side-by-side layout of strictly associated body content + margin note, such that the column widths resize appropriately as the window is resized?

**Reply by dragonstyle** [reply]
For margin notes in HTML output we really expect them to appear with the text / content that they appear in. If you have a case where this isn't happening, I'd very much like to check it out to see if I can solve that.

I'd say what we have today is really trying to be what you describe (automate side by side layout with positioning) with the exception of the grid having fixed margin widths as the document is resized (if this seems wrong to me, would love if you could say a bit more about what you're looking for). We do want to make the grid more flexible / customizable in a future update, which might help with that portion at least...
